### PR TITLE
Nav testing

### PR DESCRIPTION
**Pull request by agordon1123**
# Description

- Created tests for Nav component
- Created a function to embed analytics conditionally if environment === 'production'
- Moved analytics key to an environment variable named REACT_APP_GOOGLE_ANALYTICS_KEY
- Mounted analytics in a useEffect to call upon App mounting

### CODE REVIEW

File: mission-control-client/src/tests/Nav.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            <Router history={history}>
+                <Nav />
+            </Router>
+        );
+        // Click login button
+        await fireEvent.click(getByText('Sign In'));
+        expect(history.location.pathname).toBe('/login');
+    });
+
+    it('directs to the login page when "Sign In" is clicked', async () => {
```
Comment (danbergelt) [inline mission-control-client/src/tests/Nav.test.js:34]: This test outcome needs to be renamed @agordon1123

**Review by danbergelt — APPROVED** [review]

File: mission-control-client/src/tests/Nav.test.js (hunk 1 above)
Comment (agordon1123) [reply mission-control-client/src/tests/Nav.test.js:34]: Is it too vague? @danbergelt

File: mission-control-client/src/App.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 
-//embedded Google Analytics for web metrics in React
-ReactGA.initialize('UA-111378465-2');
-ReactGA.pageview(window.location.pathname + window.location.search);
 
 function App() {
+  
+  useEffect(() => {
+    embedAnalytics();
+  }, []);
```
Comment (armandoroman1016) [inline mission-control-client/src/App.js:17]: is a dependency needed for this function to be called more then once?

**Review by armandoroman1016 — APPROVED** [review]

File: mission-control-client/src/App.js (hunk 2 above)
Comment (agordon1123) [reply mission-control-client/src/App.js:17]: Yeah, you're right. I thought react-ga would watch the location but it does look like I'll need to pull in the useLocation hook and add it to the dependency array.

**Review by BDesselle — APPROVED** [review]

**Review by macscheffer — APPROVED** [review]

**Pull request merged**